internal/compiler/passes/const_propagation.rs

```diff
@@ -199,6 +244,30 @@ fn extract_constant_property_reference(nr: &NamedReference) -> Option<Expression
     Some(expression)
 }
 
+fn try_inline_function(function: &Callable, arguments: &[Expression]) -> Option<Expression> {
+    let Callable::Function(function) = function else {
+        return None;
+    };
+    if !function.is_constant() {
+        return None;
+    }
+    let mut body = extract_constant_property_reference(function)?;
+
+    body.visit_recursive_mut(&mut |e| {
+        if let Expression::FunctionParameterReference { index, ty } = e {
+            let Some(e_new) = arguments.get(*index).cloned() else { return };
```

Comment:
Note that sometimes function parameter can be expensive to compute.
Imagine:

```slint
// Imagine a super slow fibbonatchi function
callback fibbonatchi(x: int) -> int;
function square(x: int) -> int { return x * x; } 
out property <int> value: square(fibbonatchi(foobar));
```

If i'm not mistaken, this would expend value to 

```slint
out property <int> value: fibbonatchi(foobar) * fibbonatchi(foobar)
```

Or perhaps it's fine because we only inline if the arguments are simple?

Reply:
If the arguments contain property lookups, callbacks or any non-const function calls, the inlining doesn't happen. And even if it did for properties, the property read deduplicator would take care of it.

A sufficiently complex slint function could be problematic. In fact, a pure but recursive `slint` function would outright hang this code if it wasn't for the "binding loop" in a prior step.

So it really is a heuristic in its current form, but I couldn't really come up with a case where it behaves wrong.